File: ejemplos/get_blocknumber/rpc/get_blocknumber.py
#!/usr/bin/env python
"""Obtiene el número de bloque actual de la red Ethereum"""

from typing import List, Any

import requests

# Este script asume que hay un nodo de Ethereum en ejecución en el puerto 8545
RPC_PORT = 8545
RPC_HOST = "localhost"


def rpcreq(method: str, params: List) -> Any:
    """Realiza una llamada RPC a un nodo de Ethereum"""
    session = requests.Session()
    payload = {"jsonrpc": "2.0", "method": method, "params": params, "id": 1}
    headers = {'Content-type': 'application/json'}
    response = session.post(
        f"http://{RPC_HOST}:{RPC_PORT}",
        json=payload,
        headers=headers)
    return response.json()['result']


def get_block_number() -> int:
    """Returns the current number of the Ethereum blockchain."""
    block_number = rpcreq("eth_blockNumber", [])
    return int(block_number, 16)


if __name__ == "__main__":
    print(get_block_number())

File: ejemplos/get_blocknumber/rpc/get_blocknumber.js
#!/usr/bin/env node
import axios from 'axios'

async function rpcreq(opname, params) {
    var body = {
        jsonrpc: "2.0",
        id: 1,
        method: opname,
        params: params
    }
    let response = await axios.post(
        'http://localhost:8545',
        body
    );
    return response.data.result;
};

let blockNumber = await rpcreq('eth_blockNumber', []);
console.log(parseInt(blockNumber, 16));
